File: ruby/find_missing_number.rb
# https://www.codewars.com/kata/5a1d86dbba2a142e040000ee/train/ruby

def find_number start,stop,string
  range = (start..stop).to_a.join.chars.sort.join
  string.chars.each { |c| range.sub!(c, '') }
  p range

  (start..stop).select { |n| n.to_s.chars.sort.join == range }
end

File: python/find_missing_number.py
# https://www.codewars.com/kata/5a1d86dbba2a142e040000ee/train/ruby

def find_number(start, stop, string):
  range_str = ''.join(sorted(''.join(str(n) for n in range(start, stop + 1))))
  for c in string:
      range_str = range_str.replace(c, '', 1)
  
  print(range_str)
  return [n for n in range(start, stop + 1) if ''.join(sorted(str(n))) == range_str]
